File: VB/ObservableObject.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel

Namespace SchedulerMVVMScenarioWpf
	Public MustInherit Class ObservableObject
		Implements INotifyPropertyChanged
		Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

		Protected Sub OnPropertyChanged(ByVal propertyName As String)
			If Me.PropertyChangedEvent IsNot Nothing Then
				RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
			End If
		End Sub
	End Class
End Namespace

File: CS/ObservableObject.cs
using System;
using System.ComponentModel;

namespace SchedulerMVVMScenarioWpf {
    public abstract class ObservableObject : INotifyPropertyChanged {
        public event PropertyChangedEventHandler PropertyChanged;

        protected void OnPropertyChanged(String propertyName) {
            if (this.PropertyChanged != null)
                this.PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
        }
    }
}